nautobot_device_lifecycle_mgmt/jobs/cve_tracking.py

```diff
@@ -57,3 +71,324 @@ def run(self, published_after, debug=False):  # pylint: disable=too-many-locals,
+                    extra={"object": SoftwareVersion.objects.get(id=software_id), "grouping": "CVE Creation"},
+                )
+                cve_list = [cve["cve"] for cve in result["vulnerabilities"]]
+                dlc_cves = [cve.name for cve in CVELCM.objects.all()]
+
+                all_cve_info = self.process_cves(cve_list, dlc_cves, software_id)
+
+        return all_cve_info
+
+    def process_cves(self, cve_list, dlc_cves, software_id):
+        """Method to return processed CVE info.
+
+        Args:
+            cve_list (list): List of CVE returned from CPE search
+            dlc_cves (list): List of all DLM CVE objects
+            software_id (object): UUID of the Software being queried
+
+        Returns:
+            dict: Dictionary of CVEs in either new or existing categories
+        """
+        processed_cve_info = {"new": {}, "existing": {}}
+        if cve_list:
+            for cve in cve_list:
+                cve_name = cve["id"]
+                if cve_name.startswith("CVE"):
+                    if cve_name not in dlc_cves:
+                        processed_cve_info["new"].update({cve_name: self.prep_cve_for_dlc(cve)})
+                    else:
+                        processed_cve_info["existing"].update({cve_name: self.prep_cve_for_dlc(cve)})
+            self.logger.info(
+                "Prepared %s CVE for creation." % len(processed_cve_info["new"]),
+                extra={"object": SoftwareVersion.objects.get(id=software_id), "grouping": "CVE Creation"},
+            )
+
+        return processed_cve_info
+
+    def query_api(self, url):
+        """Establishes a session for use of retries and backoff.
+
+        Args:
+            url (string): The API endpoint getting queried.
+
+        Returns:
+            dict: Dictionary of returned results if successful.
+        """
+        try:
+            result = self.session.get(url, headers=self.headers)
+            result.raise_for_status()
+        except HTTPError as err:
+            code = err.response.status_code
+            self.logger.error(
+                "The NIST Service is currently unavailable. Status Code: %s. Try running the job again later." % code
+            )
+
+        return result.json()
+
+    @staticmethod
+    def convert_v2_base_score_to_severity(score: float) -> str:
+        """Uses V2 Base Score to convert to Severity Value.
+
+        Args:
+            score (float): CVSS V2 Base Score
+
+        Returns:
+            str: Severity Value ["HIGH", "MEDIUM", "LOW", "UNDEFINED"]
+        """
+        if 0.0 >= score <= 3.9:
+            return "LOW"
+        if 4.0 >= score <= 6.9:
+            return "MEDIUM"
+        if 7.0 >= score <= 10:
+            return "HIGH"
+        return "UNDEFINED"
+
+    def prep_cve_for_dlc(self, cve_json):  # pylint: disable=too-many-locals
+        """Converts CVE info into DLC Model compatibility."""
+        cve = cve_json
+
+        # cve_base = cve["vulnerabilities"][0]['cve']
```

Comment:
Is this still needed?

Reply:
I believe the last commit resolved this.